socorro/unittest/external/es/test_crashstorage.py

```diff
@@ -947,6 +947,34 @@ def test_crash_size_capture(self):
             assert mm.has_record('histogram', stat='processor.es.raw_crash_size', value=27)
             assert mm.has_record('histogram', stat='processor.es.processed_crash_size', value=1785)
 
+    def test_index_data_capture(self):
+        """Verify we capture index data in ES crashstorage"""
+        with MetricsMock() as mm:
+            es_storage = ESCrashStorage(config=self.config, namespace='processor.es')
+
+            mock_connection = mock.Mock()
+            # Do a successful indexing
+            es_storage._index_crash_in_elasticsearch(
+                connection=mock_connection,
+                es_index=None,
+                es_doctype=None,
+                crash_document=None,
+                crash_id=None
+            )
+            # Do a failed indexing
+            mock_connection.index.side_effect = Exception
+            with pytest.raises(Exception):
+                es_storage._index_crash_in_elasticsearch(
+                    connection=mock_connection,
+                    es_index=None,
+                    es_doctype=None,
+                    crash_document=None,
+                    crash_id=None
+                )
+
+            assert mm.has_record('histogram', stat='processor.es.successful_index')
+            assert mm.has_record('histogram', stat='processor.es.failed_index')
```

Comment:
This looks cool and is very readable.

I've read the code and it's very unlikely, but if someone rewrote it, it's possible we could end up with code that sends both timings for a success (or a fail) which would be wrong. This test wouldn't catch that. One thing you could do is assert there is one and only one `processor.es.successful_index` timing and one and only one `processor.es.failed_index` timing. You can use `.filter()` for that. Alternatively, you could split this into two separate tests and for each one make sure it has the timing it should have and doesn't have the timing it doesn't have.